packages/fxa-settings/src/components/VerifiedSessionGuard/index.tsx

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { gql, useQuery, execute } from '@apollo/client';
+import LoadingSpinner from 'fxa-react/components/LoadingSpinner';
+import AppErrorDialog from 'fxa-react/components/AppErrorDialog';
+
+const GET_SESSION = gql`
+  query GetSession {
+    session {
+      verified
+    }
+  }
+`;
+
+export const VerifiedSessionGuard = ({
```

Comment:
This component needs more work but the basic idea is to wrap things that require a verified session to view or act on with a "guard" for what to show if they're not verified